src/main/java/net/yogstation/yogbot/listeners/commands/ManageEchelonCommand.kt

```diff
@@ -0,0 +1,135 @@
+		if(args.size < 3) return DiscordUtil.reply(event, "Usage is `${args[0]} ${args[1]} <ckey>`")
+		databaseManager.byondDbConnection.use {connection -> connection.prepareStatement(
+			"""
+				SELECT `id`, `ckey`
+				FROM ${databaseManager.prefix("bound_credentials")}
+				WHERE ckey = ? AND FIND_IN_SET('allow_proxies', flags)
+			""".trimIndent()
+		).use { preparedStatement ->
+			preparedStatement.setString(1, args[2])
+			preparedStatement.executeQuery().use { results ->
+				val builder = StringBuilder("Whitelists for ckey `${args[2]}`:\n```\n")
+				val monos: MutableList<Mono<*>> = ArrayList();
+				var first = true
+				while(results.next()) {
+					builder.append("${results.getInt("id")}: Ckey ${results.getString("ckey")}\n")
+					if(builder.length > Yogbot.MAX_MESSAGE_LENGTH - 10) {
+						builder.append("```")
+						monos.add(if(first) DiscordUtil.reply(event, builder.toString()) else DiscordUtil.send(event, builder.toString()))
+						first = false
+						builder.setLength(0)
+						builder.append("```")
+					}
+				}
+				builder.append("```")
+				monos.add(if(first) DiscordUtil.reply(event, builder.toString()) else DiscordUtil.send(event, builder.toString()))
+				return Mono.`when`(monos)
+			}
+		}
+		}
+	}
+
+	private fun removeWhitelist(event: MessageCreateEvent, args: Array<String>): Mono<*> {
+		if(args.size < 3) return DiscordUtil.reply(event, "Usage is `${args[0]} ${args[1]} <whitelist>`")
+		val id = args[2].toIntOrNull() ?: return DiscordUtil.reply(event, "${args[2]} is not a valid integer")
+		val monos: MutableList<Mono<*>> = ArrayList()
+		databaseManager.byondDbConnection.use {  connection ->
+			var oldFlags = ""
+			connection.prepareStatement("""
+				SELECT `ckey`, `flags`
+				FROM ${databaseManager.prefix("bound_credentials")}
+				WHERE `id` = ? AND FIND_IN_SET('allow_proxies', flags)
+			""".trimIndent()).use { preparedStatement ->
+				preparedStatement.setInt(1, id)
+				preparedStatement.executeQuery().use {
+					if(!it.next()) return DiscordUtil.reply(event, "No binding with id $id found!")
+					monos.add(DiscordUtil.reply(event, "Deleted whitelist for ckey `${it.getString("ckey")}`"))
+					oldFlags = it.getString("flags")
+				}
+			}
+			if(oldFlags == "allow_proxies") {
+				connection.prepareStatement("""
+					DELETE FROM ${databaseManager.prefix("bound_credentials")}
+					WHERE `id` = ? AND FIND_IN_SET('allow_proxies', flags)
+				""".trimIndent()).use { preparedStatement ->
+					preparedStatement.setInt(1, id)
+					val rows = preparedStatement.executeUpdate()
+					if (rows == 0) return DiscordUtil.reply(event, "Unable to delete bond.")
+					return Mono.`when`(monos)
+				}
+			} else {
+				val flagsArray = oldFlags.split(",")
+				val newFlags = flagsArray.filter { it != "allow_proxies" }.joinToString(",")
+				connection.prepareStatement("""
+					UPDATE ${databaseManager.prefix("bound_credentials")}
+					SET flags = ?
+					WHERE `id` = ? AND FIND_IN_SET('allow_proxies', flags)
```

Comment:
Not sure if im a massive fan of the time of check / time of use discrepancy but the table isn't updated too often so im willing to let it slide